app/views/solidus_friendly_promotions/admin/shared/_number_with_currency.html.erb

```diff
@@ -0,0 +1,20 @@
+<% amount_attr ||= :amount %>
+<% currency_attr ||= :currency %>
+<% currency ||= nil %>
+<% required ||= nil %>
+
+<div is="number-with-currency" class="input-group number-with-currency <%= "number-with-currency-with-select" unless currency %> js-number-with-currency">
```

Comment:
The `is` notation only works for components that enhance existing components. In your case you want to use the component in its html tag form.

```erb
<number-with-currency class="input-group number-with-currency <%= "number-with-currency-with-select" unless currency %> js-number-with-currency">
  ...
</number-with-currency>
```